core/kernel/taint_spec.rb

```diff
@@ -44,4 +44,20 @@
       end
     end
   end
+
+  ruby_version_is "2.7"..."3.0" do
+    it "is a no-op" do
+      o = Object.new
+      o.taint
+      o.should_not.tainted?
+    end
+
+    it "warns in verbose mode" do
+      -> {
+        $VERBOSE = true
+        obj = mock("tainted")
+        obj.taint
+      }.should complain(/Object#taint is deprecated and will be removed in Ruby 3.2/)
+    end
+  end
```

Comment:
I believe `$VERBOSE ` should be rolled back to default value to not affect other specs. AFAIK it isn't done by default.

Reply:
I changed all spots to use `complain("...", verbose: true)`. I _think_ that would address this.

Reply:
Yes, `complain()` will restore it properly.
BTW I recently added a check that `$VERBOSE` isn't changed by a spec example in MSpec, just not merged upstream yet.